### PR TITLE
Help Center: Fix scroll to anchor 

### PR DESCRIPTION
**Pull request by renancarvalho**
Related to https://github.com/Automattic/dotcom-forge/issues/8359

## Proposed Changes

* There were two issues
1 - The code was trying to scroll before the content was fully loaded, added a small delay to the `setTimeout` function
2 - When navigating on the same page, the scroll was getting lost because pages with anchor were not being considered.
## Why are these changes being made?

* Fixes  https://github.com/Automattic/dotcom-forge/issues/8359

## Testing Instructions

* Use live link 
* navigate to `{livelink}/hosting-features/{yousite}`
* Make sure it navigates correctly to the proper link (anchor) when the help center is opened, more details -> https://github.com/Automattic/dotcom-forge/issues/8359
* Test regressions, like Odie/Wapuu, and other internal HelpCenter results

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: packages/help-center/src/components/help-center-article.tsx [hunk 1]
```diff
@@ -20,18 +20,23 @@ export const HelpCenterArticle = () => {
...
 	useEffect( () => {
-		//If a url includes an anchor, let's scroll this into view!
+		let timeoutId: number;
 		if ( postUrl?.includes( '#' ) && post?.content ) {
-			setTimeout( () => {
-				const anchorId = postUrl.split( '#' ).pop();
-				if ( anchorId ) {
+			const anchorId = postUrl.split( '#' ).pop();
+			if ( anchorId ) {
+				timeoutId = setTimeout( () => {
```
Comment (heavyweight) [inline packages/help-center/src/components/help-center-article.tsx:27]: The type checking fails here: ` packages/help-center/src/components/help-center-article.tsx(27,5): error TS2322: Type 'Timeout' is not assignable to type 'number'.`

https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Run_All_Unit_Tests/13156866?buildTab=log&linesState=529&logView=flowAware&expandAll=true&focusLine=567

Comment (alshakero) [reply packages/help-center/src/components/help-center-article.tsx:27]: You can do `window.setTimeout` and it will return a number.

**Review by heavyweight — APPROVED** [review]
LGTM

**Comment by escapemanuele** [comment]
Is this still needed @renancarvalho?

**Comment by renancarvalho** [comment]
Closing it for now, got obsolete and was not fully working.
c/c @escapemanuele

**Pull request closed without merging**